week03/LZW/LZW/BWT.cs

```diff
@@ -0,0 +1,119 @@
+﻿// <copyright file="BWT.cs" company="SPBU">
+// Copyright (c) Alexander Bugaev 2024. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// </copyright>
+
+namespace BurrowsWheeler;
+
+/// <summary>
+/// Class implementing Burrows-Wheeler transformation.
+/// </summary>
+public static class BWT
+{
+    private static string transformingString = string.Empty;
+
+    /// <summary>
+    /// Transforms an input string by the Burrows-Wheeler algorithm.
+    /// </summary>
+    /// <param name="inputString">String to transform.</param>
+    /// <returns>Transformed string and
+    /// the position of initial string in the rotations matrix.</returns>
+    public static (string, int) Transform(string inputString)
+    {
+        transformingString = inputString;
+        var shifts = Enumerable.Range(0, inputString.Length).ToArray();
+        Array.Sort(shifts, new RotationsComparer());
+        var position = Array.IndexOf(shifts, 0);
+        return (GetResultString(inputString, shifts), (position >= 0) ? position : 0);
+    }
+
+    /// <summary>
+    /// Reverse Burrows-Wheeler transformation.
+    /// </summary>
+    /// <param name="inputString">String transformed by BWT algorithm.</param>
+    /// <param name="position">Position of string in the rotations matrix.</param>
+    /// <returns>String after reverse transformation.</returns>
+    /// <exception cref="IndexOutOfRangeException">
+    /// Position is not in range of the input string length.</exception>
+    public static string ReverseTransform(string inputString, int position)
+    {
+        if (inputString.Length != 0 && (position < 0 || position >= inputString.Length))
+        {
+            throw new IndexOutOfRangeException();
+        }
+
+        var result = new char[inputString.Length];
+        var shifts = GetShiftsArray(inputString);
+        for (int i = 0; i < inputString.Length; ++i)
```

Comment:
```suggestion
        for (var i = 0; i < inputString.Length; ++i)
```
используй var везде, где можно :)